### DISCUSSION TITLE
What's the conceptual difference for  * buffer *  between buffer section of metadata, and c++ client's createBuffer.

### DISCUSSION THREAD

**Discussion by liusitan** [v6d-io/v6d#968] [Q&A] [answered]
Describe your problem
---------------------

If the buffer client， if I try to get a metadata of from a client, it has a buffer section
<img width="437" alt="image" src="https://user-images.githubusercontent.com/20542539/177191146-7a8505ab-69fd-4f2e-a49c-4b652285cc07.png">
while in the c++ client, it also has a createBuffer Interface
<img width="641" alt="image" src="https://user-images.githubusercontent.com/20542539/177191365-96b2fd20-41c6-42b0-9b8a-9e509376f96e.png">
are they different?

----------------------------------------------------------------------

yes, they are different, createBuffer is an internal method in the client, which will not be exposed to our user, its job is to allocate memory on the server side, and let the blob object use the memory. While the metadata's buffer section, it points to the payload of the data.

**Comment by sighingnow** [comment, marked as answer]
It is an optimize to reduce the overhead of obtaining metadata from metadata store, without traversing the blobs embeded.
